src/text.ts

```diff
@@ -1,24 +1,33 @@
-export function insertText(
-  textarea: HTMLInputElement | HTMLTextAreaElement,
-  text: string,
-  options = {addNewline: true}
-): void {
-  const beginning = textarea.value.substring(0, textarea.selectionStart || 0)
-  const remaining = textarea.value.substring(textarea.selectionEnd || 0)
+export function insertText(textarea: HTMLInputElement | HTMLTextAreaElement, text: string): void {
+  const before = textarea.value.slice(0, textarea.selectionStart || undefined)
+  const after = textarea.value.slice(textarea.selectionEnd || undefined)
 
-  const newline = !options.addNewline || beginning.length === 0 || beginning.match(/\n$/) ? '' : '\n'
-  const textBeforeCursor = beginning + newline + text
+  let canInsertText = true
 
-  textarea.value = textBeforeCursor + remaining
-  textarea.selectionStart = textBeforeCursor.length
-  textarea.selectionEnd = textarea.selectionStart
+  textarea.contentEditable = 'true'
+  try {
+    canInsertText = document.execCommand('insertText', false, text)
+  } catch (error) {
+    canInsertText = false
+  }
+  textarea.contentEditable = 'false'
 
-  textarea.dispatchEvent(
-    new CustomEvent('change', {
```

Comment:
Do you think we will run into any issues by changing from a `change` to an `input` event?

Reply:
@dgreif Good catch! I changed it back to `change` in 2c410362fa0da58f238192b4fde220053e165a3f.
I'm not sure if changing it to `input` would have any negative effect, I only copied it over accidentally from https://github.com/github/markdown-toolbar-element/blob/a5b444a0405348be2af01535a8b594147370dc4b/src/index.ts#L379-L418.

Thank you!